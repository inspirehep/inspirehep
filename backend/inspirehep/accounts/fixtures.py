# -*- coding: utf-8 -*-
#
# Copyright (C) 2019 CERN.
#
# inspirehep is free software; you can redistribute it and/or modify it under
# the terms of the MIT License; see LICENSE file for more details.


import click
from flask import current_app
from flask_security.utils import hash_password
from invenio_access.models import ActionRoles
from invenio_accounts.models import Role
from invenio_db import db
from invenio_oauth2server.models import Client, Token

from inspirehep.accounts.roles import Roles


def init_oauth_token():
    ds = current_app.extensions["invenio-accounts"].datastore
    user = ds.user_model.query.filter_by(email="admin@inspirehep.net").one()
    with db.session.begin_nested():
        client = Client(
            name="admin",
            user_id=user.id,
            is_internal=True,
            is_confidential=False,
            _default_scopes="",
        )
        client.gen_salt()

        token = Token(
            client_id=client.client_id,
            user_id=user.id,
            access_token=current_app.config["AUTHENTICATION_TOKEN"],
            expires=None,
            _scopes="",
            is_personal=True,
            is_internal=True,
        )

        db.session.add(client)
        db.session.add(token)
    db.session.commit()
    click.secho("Authentication token generated successfully", fg="green")


def init_roles():
    ds = current_app.extensions["invenio-accounts"].datastore
    with db.session.begin_nested():
        ds.create_role(
            name=Roles.superuser.value, description="admin with no restrictions"
        )
        ds.create_role(
            name=Roles.cataloger.value, description="users with editing capabilities"
        )
        ds.create_role(
            name=Roles.hermescurator.value,
            description="curator for HERMES Internal Notes",
        )
        ds.create_role(
            name=Roles.hermescoll.value,
            description="HERMES Collaboration access to Internal Notes",
        )
        ds.create_role(
            name=Roles.jlabcurator.value,
            description="curator for JLAB related articles",
        )
    db.session.commit()


def init_users():
    """Sample users, not to be used in production."""
    ds = current_app.extensions["invenio-accounts"].datastore
    superuser = Role.query.filter_by(name=Roles.superuser.value).one()
    cataloger = Role.query.filter_by(name=Roles.cataloger.value).one()
    hermes_curator = Role.query.filter_by(name=Roles.hermescurator.value).one()
    hermes_collections = Role.query.filter_by(name=Roles.hermescoll.value).one()
    jlab_curator = Role.query.filter_by(name=Roles.jlabcurator.value).one()
    with db.session.begin_nested():
        ds.create_user(
            email="admin@inspirehep.net",
            password=hash_password("123456"),
            active=True,
            roles=[superuser],
        )
        ds.create_user(
            email="cataloger@inspirehep.net",
            password=hash_password("123456"),
            active=True,
            roles=[cataloger],
        )
        ds.create_user(
            email="hermescataloger@inspirehep.net",
            password=hash_password("123456"),
            active=True,
            roles=[hermes_curator, hermes_collections],
        )
        ds.create_user(
            email="jlabcurator@inspirehep.net",
            password=hash_password("123456"),
            active=True,
            roles=[jlab_curator],
        )
        ds.create_user(
            email="johndoe@inspirehep.net",
            password=hash_password("123456"),
            active=True,
        )
    db.session.commit()


def init_superuser_permissions():
    superuser = Role.query.filter_by(name=Roles.superuser.value).one()
    db.session.add(ActionRoles(action="superuser-access", role=superuser))
    db.session.add(ActionRoles(action="admin-access", role=superuser))


def init_cataloger_permissions():
    cataloger = Role.query.filter_by(name=Roles.cataloger.value).one()
    db.session.add(ActionRoles(action="workflows-ui-admin-access", role=cataloger))
    db.session.add(ActionRoles(action="admin-holdingpen-authors", role=cataloger))
    db.session.add(ActionRoles(action="update-collection", role=cataloger))
    db.session.add(ActionRoles(action="editor-use-api", role=cataloger))
    db.session.add(ActionRoles(action="migrator-use-api", role=cataloger))


def init_hermes_permissions():
    hermes_collections = Role.query.filter_by(name=Roles.hermescoll.value).one()
    db.session.add(
        ActionRoles(
            action="view-restricted-collection",
            argument="HERMES Internal Notes",
            role=hermes_collections,
        )
    )

    hermes_curator = Role.query.filter_by(name=Roles.hermescurator.value).one()
    db.session.add(
        ActionRoles(
            action="update-collection",
            argument="HERMES Internal Notes",
            role=hermes_curator,
        )
    )


def init_jlab_permissions():
    jlab_curator = Role.query.filter_by(name=Roles.jlabcurator.value).one()
    db.session.add(ActionRoles(action="workflows-ui-read-access", role=jlab_curator))
    db.session.add(ActionRoles(action="update-collection", role=jlab_curator))


def init_permissions():
    init_superuser_permissions()
    init_cataloger_permissions()
    init_hermes_permissions()
    init_jlab_permissions()
    db.session.commit()


def init_users_and_permissions():
    init_roles()
    init_users()
    init_permissions()
