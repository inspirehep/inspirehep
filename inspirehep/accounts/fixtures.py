# -*- coding: utf-8 -*-
#
# Copyright (C) 2019 CERN.
#
# inspirehep is free software; you can redistribute it and/or modify it under
# the terms of the MIT License; see LICENSE file for more details.


import click
from flask import current_app
from flask_security.utils import hash_password
from invenio_accounts.models import Role
from invenio_db import db
from invenio_oauth2server.models import Client, Token


def init_oauth_token():
    ds = current_app.extensions["invenio-accounts"].datastore
    admin = Role.query.filter_by(name="admin").one()
    with db.session.begin_nested():
        user = ds.create_user(
            email="admin@inspirehep.net",
            password=hash_password("123456"),
            active=True,
            roles=[admin],
        )
    db.session.commit()
    click.secho("User admin added successfully", fg="green")
    with db.session.begin_nested():
        client = Client(
            name="admin",
            user_id=user.id,
            is_internal=True,
            is_confidential=False,
            _default_scopes="",
        )
        client.gen_salt()

        token = Token(
            client_id=client.client_id,
            user_id=user.id,
            access_token=current_app.config["AUTHENTICATION_TOKEN"],
            expires=None,
            _scopes="",
            is_personal=True,
            is_internal=True,
        )

        db.session.add(client)
        db.session.add(token)
    db.session.commit()
    click.secho("Authentication token generated successfully", fg="green")
