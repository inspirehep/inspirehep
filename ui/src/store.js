import { applyMiddleware, createStore as createReduxStore } from 'redux';
import thunk from 'redux-thunk';
import createHistory from 'history/createBrowserHistory';
import { routerMiddleware } from 'connected-react-router';
/* eslint-disable import/no-extraneous-dependencies */
import { createLogger } from 'redux-logger';
/* eslint-disable import/no-extraneous-dependencies */
import { composeWithDevTools } from 'redux-devtools-extension';

import createRootReducer from './reducers';
import http from './common/http';
import queryParamsParserMiddleware from './middlewares/queryParamsParser';
import keepPreviousUrlMiddleware from './middlewares/keepPreviousUrl';
import {
  reHydrateRootStateFromStorage,
  createPersistToStorageMiddleware,
} from './middlewares/statePersister';
import clearStateDispatcher from './middlewares/clearStateDispatcher';
import redirectToErrorPageMiddleware from './middlewares/redirectToErrorPage';
import actionTrackerMiddleware from './middlewares/actionTracker';
import syncLocationWithSearch from './middlewares/syncLocationWithSearch';
import logoutUserOn401 from './middlewares/logoutUserOn401';

export const thunkMiddleware = thunk.withExtraArgument(http);

export const history = createHistory();
const connectedRouterMiddleware = routerMiddleware(history);

const reducersToPersist = ['ui', 'user'];

const PROD_MIDDLEWARES = [
  connectedRouterMiddleware,
  queryParamsParserMiddleware,
  keepPreviousUrlMiddleware,
  createPersistToStorageMiddleware(reducersToPersist),
  clearStateDispatcher,
  redirectToErrorPageMiddleware,
  actionTrackerMiddleware,
  thunkMiddleware,
  syncLocationWithSearch,
  logoutUserOn401,
];

const DEV_MIDDLEWARES = [...PROD_MIDDLEWARES, createLogger()];

const withMiddlewares = () => {
  if (process.env.NODE_ENV === 'production') {
    return applyMiddleware(...PROD_MIDDLEWARES);
  }
  return applyMiddleware(...DEV_MIDDLEWARES);
};

export default function createStore() {
  return createReduxStore(
    createRootReducer(history),
    reHydrateRootStateFromStorage(reducersToPersist),
    composeWithDevTools(withMiddlewares())
  );
}
