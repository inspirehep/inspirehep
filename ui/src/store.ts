import { applyMiddleware, createStore as createReduxStore } from 'redux';
import thunk from 'redux-thunk';
// @ts-expect-error ts-migrate(7016) FIXME: Could not find a declaration file for module 'hist... Remove this comment to see the full error message
import createHistory from 'history/createBrowserHistory';
import { routerMiddleware } from 'connected-react-router';
/* eslint-disable import/no-extraneous-dependencies */
// @ts-expect-error ts-migrate(7016) FIXME: Could not find a declaration file for module 'redu... Remove this comment to see the full error message
import { createLogger } from 'redux-logger';
/* eslint-disable import/no-extraneous-dependencies */
import { composeWithDevTools } from 'redux-devtools-extension';

import createRootReducer from './reducers';
// @ts-expect-error ts-migrate(2691) FIXME: An import path cannot end with a '.ts' extension. ... Remove this comment to see the full error message
import http from './common/http.ts';
import queryParamsParserMiddleware from './middlewares/queryParamsParser';
import keepPreviousUrlMiddleware from './middlewares/keepPreviousUrl';
import {
  reHydrateRootStateFromStorage,
  createPersistToStorageMiddleware,
} from './middlewares/statePersister';
import clearStateDispatcher from './middlewares/clearStateDispatcher';
import redirectToErrorPageMiddleware from './middlewares/redirectToErrorPage';
import actionTrackerMiddleware from './middlewares/actionTracker';
import syncLocationWithSearch from './middlewares/syncLocationWithSearch';
import logoutUserOn401 from './middlewares/logoutUserOn401';

export const thunkMiddleware = thunk.withExtraArgument(http);

export const history = createHistory();
const connectedRouterMiddleware = routerMiddleware(history);

const reducersToPersist = ['ui', 'user'];

const PROD_MIDDLEWARES = [
  connectedRouterMiddleware,
  queryParamsParserMiddleware,
  keepPreviousUrlMiddleware,
  // @ts-expect-error ts-migrate(2554) FIXME: Expected 0 arguments, but got 1.
  createPersistToStorageMiddleware(reducersToPersist),
  clearStateDispatcher,
  redirectToErrorPageMiddleware,
  actionTrackerMiddleware,
  thunkMiddleware,
  syncLocationWithSearch,
  logoutUserOn401,
];

const DEV_MIDDLEWARES = [...PROD_MIDDLEWARES, createLogger()];

const withMiddlewares = () => {
  if (process.env.NODE_ENV === 'production') {
    return applyMiddleware(...PROD_MIDDLEWARES);
  }
  return applyMiddleware(...DEV_MIDDLEWARES);
};

export default function createStore() {
  return createReduxStore(
    createRootReducer(history),
    // @ts-expect-error ts-migrate(2554) FIXME: Expected 0 arguments, but got 1.
    reHydrateRootStateFromStorage(reducersToPersist),
    composeWithDevTools(withMiddlewares())
  );
}
