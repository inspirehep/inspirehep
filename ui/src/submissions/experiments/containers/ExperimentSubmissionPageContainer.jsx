import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Map } from 'immutable';
import { Link } from 'react-router-dom';

import { submit } from '../../../actions/submissions';
import SubmissionPage from '../../common/components/SubmissionPage';
import ExperimentSubmission from '../components/ExperimentSubmission';
import { EXPERIMENTS_PID_TYPE } from '../../../common/constants';
import { EXPERIMENTS } from '../../../common/routes';

export const ExperimentSubmissionPage = ({ error, onSubmit }) => (
  <SubmissionPage
    title="Suggest experiment"
    description={
      <span>
        This form allows you to create a new experiment record. It will appear
        in the <Link to={EXPERIMENTS}>Experiments List</Link> immediately.
      </span>
    }
  >
    <ExperimentSubmission error={error} onSubmit={onSubmit} />
  </SubmissionPage>
);

ExperimentSubmissionPage.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  error: PropTypes.instanceOf(Map),
};

const stateToProps = (state) => ({
  error: state.submissions.get('submitError'),
});

const dispatchToProps = (dispatch) => ({
  async onSubmit(formData) {
    await dispatch(submit(EXPERIMENTS_PID_TYPE, formData));
  },
});

export default connect(stateToProps, dispatchToProps)(ExperimentSubmissionPage);
