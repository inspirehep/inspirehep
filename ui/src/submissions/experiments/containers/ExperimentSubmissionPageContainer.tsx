import React from 'react';
import { connect, RootStateOrAny } from 'react-redux';
import { Action, ActionCreator } from 'redux';
import { Map } from 'immutable';
import { Link } from 'react-router-dom';

import { submit } from '../../../actions/submissions';
import SubmissionPage from '../../common/components/SubmissionPage';
import ExperimentSubmission from '../components/ExperimentSubmission';
import { EXPERIMENTS_PID_TYPE } from '../../../common/constants';
import { EXPERIMENTS } from '../../../common/routes';

export const ExperimentSubmissionPage = ({
  error,
  onSubmit,
}: {
  error: Map<string, any>;
  onSubmit: Function;
}) => (
  <SubmissionPage
    title="Suggest experiment"
    description={
      <span>
        This form allows you to create a new experiment record. It will appear
        in the <Link to={EXPERIMENTS}>Experiments List</Link> immediately.
      </span>
    }
  >
    <ExperimentSubmission error={error} onSubmit={onSubmit} />
  </SubmissionPage>
);

const stateToProps = (state: RootStateOrAny) => ({
  error: state.submissions.get('submitError'),
});

const dispatchToProps = (dispatch: ActionCreator<Action>) => ({
  async onSubmit(formData: any) {
    await dispatch(submit(EXPERIMENTS_PID_TYPE, formData));
  },
});

export default connect(stateToProps, dispatchToProps)(ExperimentSubmissionPage);
