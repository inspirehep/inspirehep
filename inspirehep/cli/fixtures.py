# -*- coding: utf-8 -*-
#
# Copyright (C) 2019 CERN.
#
# inspirehep is free software; you can redistribute it and/or modify it under
# the terms of the MIT License; see LICENSE file for more details.

import click
from flask.cli import with_appcontext

from inspirehep.accounts.fixtures import init_oauth_token
from inspirehep.records.fixtures import (
    init_default_storage_path,
    init_records_files_storage_path,
)


@click.group()
def fixtures():
    """Command related to records in inspire"""


@fixtures.command()
@with_appcontext
def init():
    init_default_storage_path()
    init_records_files_storage_path()
    init_oauth_token()
