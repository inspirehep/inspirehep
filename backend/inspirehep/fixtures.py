# -*- coding: utf-8 -*-
#
# Copyright (C) 2019 CERN.
#
# inspirehep is free software; you can redistribute it and/or modify it under
# the terms of the MIT License; see LICENSE file for more details.

import click
from flask.cli import with_appcontext

from inspirehep.accounts.fixtures import init_oauth_token, init_users_and_permissions


@click.group()
def fixtures():
    """Command related to records in inspire"""


@fixtures.command()
@with_appcontext
def init():
    init_users_and_permissions()
    init_oauth_token()
