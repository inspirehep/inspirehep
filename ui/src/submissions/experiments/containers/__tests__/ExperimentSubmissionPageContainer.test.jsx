import React from 'react';
import { mount, shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { fromJS } from 'immutable';

import ExperimentSubmissionPageContainer, {
  ExperimentSubmissionPage,
} from '../ExperimentSubmissionPageContainer';

import { getStoreWithState } from '../../../../fixtures/store';

describe('ExperimentSubmissionSuccessPageContainer', () => {
  it('passes props to ExperimentSubmissionSucessPage', () => {
    const store = getStoreWithState({
      submissions: fromJS({
        submitError: null,
      }),
    });
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <ExperimentSubmissionPageContainer />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.find(ExperimentSubmissionPage)).toHaveProp({
      error: null,
    });
  });

  describe('ExperimentSubmissionSucessPage', () => {
    it('renders', () => {
      const component = shallow(
        <ExperimentSubmissionPage error={null} onSubmit={() => {}} />
      );
      expect(component).toMatchSnapshot();
    });
  });
});
