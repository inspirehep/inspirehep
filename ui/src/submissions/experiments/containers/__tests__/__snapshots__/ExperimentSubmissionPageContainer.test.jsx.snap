// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperimentSubmissionSuccessPageContainer ExperimentSubmissionSucessPage renders 1`] = `
<SubmissionPage
  description={
    <span>
      This form allows you to create a new experiment record. It will appear in the 
      <ForwardRef
        to="/experiments"
      >
        Experiments List
      </ForwardRef>
       immediately.
    </span>
  }
  title="Suggest experiment"
>
  <toAllJS(ExperimentSubmission)
    error={null}
    onSubmit={[Function]}
  />
</SubmissionPage>
`;
